portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteApps.testcase

```diff
@@ -49,10 +49,27 @@ definition {
 		property portal.acceptance = "true";
 
 		task ("Create Vanilla Counter as a Custom Element") {
-			JSONRemoteApp.addCustomElementRemoteAppEntry(
-				customElementHTMLName = "vanilla-counter",
-				customElementName = "Vanilla Counter",
-				customElementURL = "http://remote-component-test.wincent.com/packages/vanilla-counter/index.js");
+			RemoteApps.goToRemoteAppsPortlet();
+
+			RemoteApps.addCustomElementAllFields(
+				entryCSSURL = "http://remote-component-test.wincent.com/index.css",
+				entryHTMLName = "vanilla-counter",
+				entryName = "Vanilla Counter",
+				entryProperties = "test-data-user=QAuser",
+				entryURL = "http://remote-component-test.wincent.com/packages/vanilla-counter/index.js");
+		}
+
+		task ("Assert that the Vanilla Counter fields have been saved correctly") {
+			Click(
+				key_tableEntryName = "Vanilla Counter",
+				locator1 = "RemoteApps#TABLE_ENTRY_NAME_REMOTE_TABLE");
+
+			RemoteApps.assertCustomElementFields(
+				entryCSSURL = "http://remote-component-test.wincent.com/index.css",
+				entryHTMLName = "vanilla-counter",
+				entryName = "Vanilla Counter",
+				entryProperties = "test-data-user=QAuser",
+				entryURL = "http://remote-component-test.wincent.com/packages/vanilla-counter/index.js");
```

Comment:
Move this task after "Assert that the Vanilla Counter is listed". It would make better sense from UI/user flow to validate that the entry is listed first and then click into the entry and check the fields.

Reply:
`https://liferay.github.io/liferay-frontend-projects/vanilla-counter/index.js`